graph/src/components/store.rs

```diff
@@ -397,19 +415,255 @@ pub trait Store: Send + Sync + 'static {
+        let current_version_id = match subgraph_entity
+            .get("currentVersion")
+            .ok_or_else(|| format_err!("Subgraph entity without `currentVersion`"))?
+        {
+            Value::String(s) => s.to_owned(),
+            Value::Null => return Ok(None),
+            _ => {
+                return Err(format_err!(
+                    "Subgraph entity has wrong type in `currentVersion`"
+                ));
+            }
+        };
+
+        // Read subgraph version entity
+        let version_entity_opt = self
+            .get(SubgraphVersionEntity::key(current_version_id))
+            .map_err(QueryError::from)?;
+        if version_entity_opt == None {
+            return Ok(None);
+        }
+        let version_entity = version_entity_opt.unwrap();
+
+        // Parse subgraph ID
+        let subgraph_id_str = version_entity
+            .get("deployment")
+            .ok_or_else(|| format_err!("SubgraphVersion entity without `deployment`"))?
+            .to_owned()
+            .as_string()
+            .ok_or_else(|| format_err!("SubgraphVersion entity has wrong type in `deployment`"))?;
+        SubgraphDeploymentId::new(subgraph_id_str)
+            .map_err(|()| {
+                format_err!("SubgraphVersion entity has invalid subgraph ID in `deployment`")
+            })
+            .map(Some)
+    }
+
+    /// Read all version entities pointing to the specified deployment IDs and determine whether
+    /// they are current or pending in order to produce `SubgraphVersionSummary`s.
+    fn read_subgraph_version_summaries(
+        &self,
+        deployment_ids: Vec<SubgraphDeploymentId>,
+    ) -> Result<(Vec<SubgraphVersionSummary>, Vec<EntityOperation>), Error> {
+        let version_filter = EntityFilter::new_in(
+            "deployment",
+            deployment_ids.iter().map(|id| id.to_string()).collect(),
+        );
+
+        let mut ops = vec![];
+
+        let versions = self.find(SubgraphVersionEntity::query().filter(version_filter.clone()))?;
+        let version_ids = versions
+            .iter()
+            .map(|version_entity| version_entity.id().unwrap())
+            .collect::<Vec<_>>();
+        ops.push(EntityOperation::AbortUnless {
+            description: "Same set of subgraph versions must match filter".to_owned(),
+            query: SubgraphVersionEntity::query().filter(version_filter),
+            entity_ids: version_ids.clone(),
+        });
+
+        // Find subgraphs with one of these versions as current or pending
+        let subgraphs_with_version_as_current_or_pending =
+            self.find(SubgraphEntity::query().filter(EntityFilter::Or(vec![
+                EntityFilter::new_in("currentVersion", version_ids.clone()),
+                EntityFilter::new_in("pendingVersion", version_ids.clone()),
+            ])))?;
+        let subgraph_ids_with_version_as_current_or_pending =
+            subgraphs_with_version_as_current_or_pending
```

Comment:
It would be nice to have a helper for all these transformations. Something that takes an `IntoIterator<T>` and a closure from T to U, maps over the iterator using the closure, calls collect and returns a `Coll<U>` that can be inferred from the variable type the result gets assigned to. Something like:
```
let ids: HasSet<_> = map(versions, |version| version.id().to_string()); 
```

But that's just a general thought, not important for this PR.

Reply:
I wonder if @leodasvacas knows any stdlib tricks for shortening this up that I'm not using? It does get annoying having to `iter()` and `collect()` so much...